packages/data/addon/utils/search.ts

```diff
@@ -74,25 +75,31 @@ export default {
    *          record - asset record
    *          relevance - distance between record and search query
    */
-  searchDimensionRecords(records, query, resultLimit, page) {
+  searchDimensionRecords(
+    records: NativeArray<unknown>,
+    query: string,
+    resultLimit: number,
+    page = 1
+  ): { record: Record<string, string>; relevance: number }[] {
     let results = [],
       record;
 
     // Filter, map, and sort records based on how close each record is to the search query
     for (let i = 0; i < records.length; i++) {
-      record = records.objectAt(i);
+      record = records.objectAt(i) as Record<string, string>;
 
       // Determine relevance based on string match weight
-      let descriptionMatchWeight = this.getPartialMatchWeight(
-          (record.description || '').toLowerCase(),
+      const { id, ...rest } = record;
```

Comment:
Even in just the fili context `id` does not have always exist. Can we maybe pass `'id', ['desc', 'other']` as parameters based on the columnMetadata?

Reply:
Unfortunately we don't really have access to meta data here, we'll need to probably come up with an enhanced version later.
But this code does handle the fact some dimensions don't have 'id' It will get picked up and weighted with the ...rest fields and id field weighting will be ignored.